realm/realm-library/src/main/java/io/realm/sync/permissions/RealmPrivileges.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.sync.permissions;
+
+import io.realm.Realm;
+import io.realm.RealmModel;
+import io.realm.internal.annotations.ObjectServer;
+
+/**
+ * This object describes the privileges granted the currently logged in user for either the Realm,
+ * a specific Realm Model Class or an Object. Some of the privileges are not applicable at the
+ * different levels, so see the individual privileges for the details.
+ * <p>
+ * It is obtained by calling either {@link Realm#getPrivileges()}, {@link Realm#getPrivileges(Class)}
+ * or {@link Realm#getPrivileges(RealmModel)} .
+ */
+@ObjectServer
+public final class RealmPrivileges {
+
+    private boolean canRead;
+    private boolean canUpdate;
+    private boolean canDelete;
+    private boolean canSetPermissions;
+    private boolean canQuery;
+    private boolean canCreate;
+    private boolean canModifySchema;
+
+    public RealmPrivileges(long privileges) {
+        this.canRead = (privileges & (1 << 0)) != 0;
+        this.canUpdate = (privileges & (1 << 1)) != 0;
+        this.canDelete = (privileges & (1 << 2)) != 0;
+        this.canSetPermissions = (privileges & (1 << 3)) != 0;
+        this.canQuery = (privileges & (1 << 4)) != 0;
+        this.canCreate = (privileges & (1 << 5)) != 0;
+        this.canModifySchema = (privileges & (1 << 6)) != 0;
+    }
+
+    /**
```

Comment:
Doc comments in this file seem copy-pasted — I won't repeat the review feedback here, but the same comments apply. :)